### PR TITLE
Pull request BurkinaFaso 2017-18 JW 20210605

### PR DESCRIPTION
**Pull request by jianingwwww**
- change file names from DHS to MIS
- delete rename lines
- generate survey-specific year variable
- replace country name to to align with the name in iso.dta

### CODE REVIEW

**Comment by wengxyu1030** [comment]
Thanks for the input @imarabbit. Overall it looks good. However, could you check again on the value c_anc_ski? Comparing to the DHS public data c_anc_ski at 80.2, the value produced by the code is too large at 99.3. I would suggest you might want to look into the population that the calculation based on? Or the missing value issue? 

image

**Comment by robin-wang** [comment]
Suggest checking inputs of c_anc_ski, paying attention to missing values in the process as well as value change along the way.
It might be really helpful to unearth whether certain values have been forced into missing.

Consider using pause on to validate the changes by pausing the running program in question.
@imarabbit

**Comment by jianingwwww** [comment]
Thank you @wengxyu1030 @robin-wang for your suggestions. I have checked the places you mentioned and here are my findings:

The reason why c_anc_ski is so large is that anc_skill used to generate it has too many 0 values. Therefore, I further checked the generation process of anc_skill (i.e., how did we generate anc_skill from m2a-m2m).

The variables with actual values in m2a-m2m (i.e., not na) are: 1) doctor; 2) nurse; 3) midwife; 4) patented birth attendant / auxiliary birth attendant; 5) skilled birth attendant; 6) traditional /village birth attendant; 7) community health worker; 8) other. The first five are skilled providers and the last three are non-skilled.

According to the logic of the code, the first five categories will be retained at the end (these five skilled providers can correspond to the five PROFESSIONNEL DE SANTÉ options in the survey questionnaire), and the last three non-skilled categories will be converted to missing values (no problem here either). It seems that the code works without problems.

To double-check it, I used the "count if" command and found that the number of 0 value was indeed 35. (In my understanding, anc_ski=0 means that the five variables of skilled provider are 0 at the same time, which also means that no antenatal care has been received from any of these five types of skilled providers. )The result of "count if" shows that there are only 35 cases where all five variables are 0 at the same time, which corresponds to the previous result of an abnormally low value of 0.

In summary, I prefer that this might be a problem with the dataset itself rather than with the code.

**Comment by robin-wang** [comment]
Hi @imarabbit Thank you for your thoughtful comments. That is quite consistent with what we discussed earlier today, I appreciate you organising them here for future reference.

There might be further revision to how we process raw data earlier on in the pipeline, but for the moment, let's move on to Burundi.

**Comment by wengxyu1030** [comment]
I agree with the decision. The time-series checking later could also tackle this issue.

**Comment by Xian152** [comment]
Hi all @wengxyu1030 @robin-wang @jianingwwww 
I took a quick look at this issue. First, it seems that the report doesn't give us their definition of skilled provider, so we have to judge which providers should be classified skilled 
If we refer to our list of skilled provider, we would regard 1) doctor; 2) nurse; 3) midwife; 4) patented birth attendant / auxiliary birth attendant; 5) skilled birth attendant; as skilled, just like @jianingwwww pointed out. Not 100% confident about "patented birth attendant", but usually "auxiliary birth attendant" are regarded as skilled.
However, sometimes we have to do deduction backward and guess how DHS judege who is skilled provider for this survey. If we look at the share 
  Variable |        Obs        Mean    Std. Dev.       Min        Max
-------------+---------------------------------------------------------
         m2a |      4,517    .0307726    .1727203          0          1
         m2b |      4,517    .3114899    .4631538          0          1
         m2c |      4,517    .5691831    .4952454          0          1
         m2d |      4,517    .1941554    .3955929          0          1
         m2e |      4,517    .0208103    .1427646          0          1
-------------+---------------------------------------------------------
I am guessing DHS regard m2d as not skilled while we does. 

But there is a curious thing: When doing quality check, my results only report 2 variables for this survey. I wonder where does QC result for c_anc_ski come from?
Screen Shot 2021-06-19 at 1 56 48 AM

**Comment by robin-wang** [comment]
20210622 RW Comments:
This is merged, but we still need to work out a few things
1. why Burkina Faso's Quality Checking output seems to have more variables than Xian indicated
2. Burkina Faso's coding on skilled providers, might we try sensitivity analysis to see if we could approach the DHS values (backward engineering literally...)
3. Raw data in Burkina Faso 2017-18 folder is now suffixed with 2014.

**Comment by jianingwwww** [comment]
Hi @robin-wang @Xian152, thank you for your comments! Here are my responses and findings:

1. For now I can not find out why there are only 2 variables on Xian's end. I have run the codes (MAIN branch) on my end and it looks fine.
83f7ada9d580462e368df1a4893d418

2. The reason I considered auxiliary birth attendant as a skilled provider was the classification of the questionnaire itself (I thought PROFESSIONNEL DE SANTÉ was similar to skilled provider). 
95471a55455e4207f6cadd1a7eb55ec
But you are right, that's might not be how DHS defines skilled provider. I have conducted the sensitivity analysis and I found that when I replace m2d with missing values, our micro data can align with the DHS public data and pass the quality check : ) So should I uploaded new micro data and make new commits?
3f853d65c0eca52fd9ef43c75fd297d




3. The raw data I was given was the same, but I found out that it was just a naming issue and the data itself was still 17-18 instead of 14. I double-checked it after I had access to the DHS database and everything looks good. I've just uploaded renamed data so you can find the new one in the raw data folder in OneDrive now. 

Hope this reply can explain!

**Comment by robin-wang** [comment]
@jianingwwww Great!

1. No action needed.
2. Let's upload the updated veresion to align with DHS for now. 
3. Could you please make sure the raw data files in the following data folder are updated please? Cheers!
https://worldbankgroup-my.sharepoint.com/:f:/r/personal/sneelsen_worldbank_org/Documents/Projects/UHC/04%20STC/Aline/World%20Bank/MEASURE%20UHC%20DATA/RAW%20DATA/MIS/MIS-BurkinaFaso2017-18?csf=1&web=1&e=TlzxgS

**Comment by jianingwwww** [comment]
> @jianingwwww Great!
> 
> 1. No action needed.
> 2. Let's upload the updated veresion to align with DHS for now.
> 3. Could you please make sure the raw data files in the following data folder are updated please? Cheers!
>    https://worldbankgroup-my.sharepoint.com/:f:/r/personal/sneelsen_worldbank_org/Documents/Projects/UHC/04%20STC/Aline/World%20Bank/MEASURE%20UHC%20DATA/RAW%20DATA/MIS/MIS-BurkinaFaso2017-18?csf=1&web=1&e=TlzxgS

2. I have pulled a new request and referred to this one. Updated micro data has also been uploaded to OneDrive.
3. This raw data file is exactly the updated version~
